commons/src/main/java/org/infinispan/commons/marshall/SerializeWith.java

```diff
@@ -33,4 +33,10 @@
     */
    Class<? extends Externalizer<?>> value();
 
+   /**
+    * TODO
+    * @return
+    */
+   SerializeValueMatcher valueMatcher() default SerializeValueMatcher.MATCH_ALWAYS;
```

Comment:
Maybe the function to execute when retrying could return a boolean success value, so if it returns `true` we consider that it did update the value (and replicate the update to the backup owners)?

Note that the backups should never consider the previous value, even if they're conditional. Delta updates need to read the previous value, and this actually works ok for AtomicMap because its delta updates are still idempotent. But a non-idempotent Delta (like the DeltaAwareList used for M/R) would create inconsistencies.